stable_baselines3/common/preprocessing.py

```diff
@@ -155,7 +156,7 @@ def get_obs_shape(
         else:
             return (int(observation_space.n),)
     elif isinstance(observation_space, spaces.Dict):
-        return {key: get_obs_shape(subspace) for (key, subspace) in observation_space.spaces.items()}
```

Comment:
What was the error here?

Reply:
because it is doing a recursive call, it assumes that the results might be incorrect as it doesn't know we don't allow nested dict.